src/main/java/app/controllers/PresentationController.java

```diff
@@ -39,12 +56,59 @@ public String presentation(@PathVariable Long id,
         }
         return times;
     }
+    @PostMapping("/project/{pid}/presentation/update")
+    public String updateTimeSlots(@PathVariable Long pid,
+                                  @RequestParam("timeSlot") String id,
+                                  @AuthenticationPrincipal UserDetails currentUser) {
+
+        System.out.println();
+        System.out.println();
+        System.out.println();
+        System.out.println();
+        if(id.split(",").length > 1) {
+            System.out.println("skipping");
+            return "redirect:/project/" + pid + "/presentation";
+        }
+        AuthenticatedUser authenticatedUser = authenticatedUserService.findByUsername(currentUser.getUsername());
+        User user;
+        Project proj = projectRepository.findOne(pid);
+        if(authenticatedUser.getType().equals("Student")) {
+            System.out.println("student");
+            user = authenticatedUser.getStudent();
+            Student stud = (Student) user;
+            if(stud.getProject().equals(proj)) {
+                System.out.println("about to update");
+                updateTimes(id, proj);
+            }
+        } else {
+            System.out.println("professor");
+            user = authenticatedUser.getProfessor();
+            Professor prof = (Professor) user;
+            if(prof.getProjects().contains(proj) || prof.getSecondReaderProjects().contains(proj)) {
+                System.out.println("about to update");
+                updateTimes(id, proj);
+            }
+        }
+        return "redirect:/project/" + pid + "/presentation";
+    }
     @PostMapping("/project/{id}/presentation/new-time")
     public String newTimeSlot(@PathVariable Long id,
                               @RequestParam("day") String day,
                               @RequestParam("hour") String hour,
-                              @RequestParam("minute") String minute) {
+                              @RequestParam("minute") String minute,
+                                @AuthenticationPrincipal UserDetails currentUser) {
+        AuthenticatedUser authenticatedUser = authenticatedUserService.findByUsername(currentUser.getUsername());
+        User user;
         Project project = projectRepository.findById(id);
+        if(!authenticatedUser.getType().equals("Student")) {
```

Comment:
I would change this to `if(authenticatedUser.getType().equals("Student")) return "redirect:/studentMenu";` Then you can get rid of the else block and have everything else follow after.